kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/Utils.java

```diff
@@ -57,6 +61,9 @@
   public static final String PATH_WINDOWS = "Path";
   public static final String PATH_UNIX = "PATH";
   private static final Random random = new Random();
+  
+  private static final ExecutorService SHARED_POOL = Executors.newCachedThreadPool();
```

Comment:
Thread creation in Cached Thread Pools are probably lazy by default (threads are only created when needed and then reused. However, I'm not sure if eagerly initializing this variable might have other consequences when users are not using any of the features that consume this pool.

Reply:
> threads are only created when needed and then reused

That is correct.

> However, I'm not sure if eagerly initializing this variable might have other consequences when users are not using any of the features that consume this pool.

It will not, but it can of course be instantiated on demand if you want.

Reply:
> It will not, but it can of course be instantiated on demand if you want.

Not sure if it's worth the effort

This was more a disclaimer comment ;P